### PR TITLE
Event documentation blog article

### PR DESCRIPTION
**Pull request by ja0x**

### CODE REVIEW

File: content/blog/events-documentation.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Slug: new-events-documentation
+Status: published
+
+Hello Wazo community!  
+
+In line with launching our new [event documentation](https://wazo-platform.org/documentation) section on our website, in the last few months, 
+we have been busy refactoring our events subsystem in Wazo to make it simpler for developers and ourselves.  
+
+The first change, and most important, is that events are now fully _tenant-aware_, meaning they will never be dispatched to users outside
+of it's intended tenant.
```
Comment (Romej) [inline content/blog/events-documentation.md:15]:
```suggestion
of its intended tenant.
```

File: content/blog/events-documentation.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Meant to be consumed by all connected users of a tenant.
+These events usually represent changes at the tenant level on the stack (i.e: configuration setting changed)
+
+#### User-Level Events:
+Meant to be consumed by individual users.  
+These events usually represent specific actions targeting a user (i.e: joining a meeting or a conference, 
+receiving a chat message, receiving a call, etc)
+
+
+At a more technical level, here is how we implement it within wazo:
```
Comment (Romej) [inline content/blog/events-documentation.md:32]:
```suggestion
At a more technical level, here is how we implement it within Wazo:
```

File: content/blog/events-documentation.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+For an event to be forwarded to a user connected through the websocket, the event must meet the following criteria:
+* Headers must have an entry `tenant_uuid` = <tenant uuid>
+* Headers must have an entry `user_uuid:<user uuid>` = true (or `user_uuid:*` = true for all users) 
+
+Any event will always be available to services, but to be relayed to users, these entries are mandatory.
+
+# Sounds cool, but how does it work in practice
+
+Here’s a coding example of how we recommend we define, for instance a new user event
```
Comment (Romej) [inline content/blog/events-documentation.md:42]:
```suggestion
Here’s a coding example of how we recommend defining, for instance, a new user event
```

File: content/blog/events-documentation.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	‘user_uuid:731e36bc-5104-4539-b0b2-bb38932dfc16’: True,
+	[…]
+}
+```
+
+
+# What if I need **all** events?
+
+In the old system, it was possible for a user to have access to all events (as long as it had the correct permissions).  
+But what if I still need that behavior, i.e in a m2m (machine to machine) forwarding/dispatching scenario?
```
Comment (Romej) [inline content/blog/events-documentation.md:83]:
```suggestion
But what if I still need that behavior, e.g in a m2m (machine to machine) forwarding/dispatching scenario?
```
If the goal is to present an example, use `e.g` (exempli gratia)
`i.e` is more like "that is" or "c'est à dire"

**Review by Romej — COMMENTED** [review]
Great read 📖

File: content/blog/events-documentation.md (hunk 4 above)
Comment (ja0x) [reply content/blog/events-documentation.md:83]: Nice, I never really read on that, thanks

File: content/blog/events-documentation.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
+Title: New events documentation
```
Comment (chartrandf) [inline content/blog/events-documentation.md:1]: Title case
```suggestion
Title: New Events Documentation
```

File: content/blog/events-documentation.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+#### Tenant-Level Events:
+Meant to be consumed by all connected users of a tenant.
+These events usually represent changes at the tenant level on the stack (i.e: configuration setting changed)
+
+#### User-Level Events:
+Meant to be consumed by individual users.  
+These events usually represent specific actions targeting a user (i.e: joining a meeting or a conference, 
+receiving a chat message, receiving a call, etc)
+
```
Comment (chartrandf) [inline content/blog/events-documentation.md:30]: (SEO tips) Maybe we should convert to list. H4 in the middle of an article in note recommanded. We should follow title increment (h1, h2, h3, h4)

```suggestion
- **Service-Level Events:** Meant to be consumed by internal services (and plugins).  Will never be received by users.
- **Tenant-Level Events:** Meant to be consumed by all connected users of a tenant. These events usually represent changes at the tenant level on the stack (i.e: configuration setting changed)
- **User-Level Events:** Meant to be consumed by individual users. These events usually represent specific actions targeting a user (i.e: joining a meeting or a conference, receiving a chat message, receiving a call, etc)

```

File: content/blog/events-documentation.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+At a more technical level, here is how we implement it within Wazo:
+
+For an event to be forwarded to a user connected through the websocket, the event must meet the following criteria:
+* Headers must have an entry `tenant_uuid` = <tenant uuid>
+* Headers must have an entry `user_uuid:<user uuid>` = true (or `user_uuid:*` = true for all users) 
+
+Any event will always be available to services, but to be relayed to users, these entries are mandatory.
+
+# Sounds cool, but how does it work in practice
```
Comment (chartrandf) [inline content/blog/events-documentation.md:40]: (SEO tips) convert to H2 since H1 is the title of the article
```suggestion
## Sounds cool, but how does it work in practice
```

File: content/blog/events-documentation.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+In the old system, it was possible for a user to have access to all events (as long as it had the correct permissions).  
+But what if I still need that behavior, e.g in a m2m (machine to machine) forwarding/dispatching scenario?
+
+Well, we have you covered!
+
+If a user belongs to the stack's _master tenant_ (in opposition to a regular tenant), from the bus perspective, that user 
+is considered an admin and will be able to receive all events it is registered to, independently of headers.
+ 
+
+# TLDR
```
Comment (chartrandf) [inline content/blog/events-documentation.md:91]: Ditto
```suggestion
## TLDR
```

File: content/blog/events-documentation.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+* _name_: name of the event, used for routing the message
+* _routing_key_fmt_: routing key used to route messages in a topic exchange (compatibility with older version)
+
+Now, when we publish this event using our publisher, headers will be generated automatically. 
+In our example, because our event is derived from `UserEvent`, the headers will be (using example values): 
+```py
+{
+	‘name’: ‘do_something’,
+	‘tenant_uuid’: 57b7080f-1d39-4269-af22-4df0f4ac6112,
+	‘user_uuid:731e36bc-5104-4539-b0b2-bb38932dfc16’: True,
```
Comment (chartrandf) [inline content/blog/events-documentation.md:74]: `users_uuid` value since not valid

**Review by chartrandf — COMMENTED** [review]
Nice works Julien :) Small suggestions for the structure of titles mostly

**Review by JipiLambert — APPROVED** [review]
